transport/stream_test.go

```diff
@@ -1,20 +1,67 @@
 package transport
 
 import (
-	"math/rand"
-
 	"scylla-go-driver/frame"
+	"testing"
 )
 
-// TestStreamIDAllocator is StreamIDAllocator that returns a random.
-type TestStreamIDAllocator struct{}
+func TestDefaultStreamAllocatorAlloc(t *testing.T) {
+	t.Parallel()
+	s := streamIDAllocator{}
+
+	allocated := make(map[frame.StreamID]struct{})
+
+	// Check if allocating all possible non-negative streams is possible.
+	for i := 0; i <= maxStreamID; i++ {
+		stream, err := s.Alloc()
+		if err != nil {
+			t.Fatalf("unable to get stream %d", i)
+		}
+
+		if _, ok := allocated[stream]; ok {
+			t.Fatalf("got an already allocated stream %d", stream)
+		}
+
+		if stream != frame.StreamID(i) {
+			t.Fatalf("expected stream %d, got stream %d", i, stream)
+		}
 
-var _ StreamIDAllocator = TestStreamIDAllocator{}
+		allocated[stream] = struct{}{}
+	}
 
-func (t TestStreamIDAllocator) Alloc() (frame.StreamID, error) {
-	const mask = int32(0xFFFF)
-	return frame.StreamID(rand.Int31() & mask), nil
+	// All streams are taken, we shouldn't be able to Alloc() another.
+	if _, err := s.Alloc(); err == nil {
+		t.Fatalf("allocating more than maxStreamID + 1 times in a row should fail")
+	}
 }
 
-func (t TestStreamIDAllocator) Free(id frame.StreamID) {
+func TestDefaultStreamAllocatorFree(t *testing.T) {
```

Comment:
It seems you could have a single test here. You can split them to different phases with subsets if you feel like separating them a little.